sdks/python/apache_beam/ml/inference/base.py

```diff
@@ -82,6 +82,33 @@ def _to_microseconds(time_ns: int) -> int:
   return int(time_ns / _NANOSECOND_TO_MICROSECOND)
 
 
+def _convert_to_result(
+    batch: Iterable,
+    predictions: Union[Iterable, Dict[Any, Iterable]],
+    drop_example: bool = False) -> Iterable[PredictionResult]:
+  if isinstance(predictions, dict):
+    # Go from one dictionary of type: {key_type1: Iterable<val_type1>,
+    # key_type2: Iterable<val_type2>, ...} where each Iterable is of
+    # length batch_size, to a list of dictionaries:
+    # [{key_type1: value_type1, key_type2: value_type2}]
+    predictions_per_tensor = [
+        dict(zip(predictions.keys(), v)) for v in zip(*predictions.values())
+    ]
+    if not drop_example:
+      return [
+          PredictionResult(x, y) for x, y in zip(batch, predictions_per_tensor)
+      ]
+    else:
+      return [
+          PredictionResult(None, y) for x,
+          y in zip(batch, predictions_per_tensor)
+      ]
```

Comment:
Rather than returning from inside this nested loop, could we just set `predictions = [dict(zip(predictions.keys(), v)) for v in zip(*predictions.values())]` and let the if/else handle it below? That way we don't need to duplicate logic

Reply:
Done. PTAL